src/ConstrainedPhotometryModel.cc

```diff
@@ -237,24 +252,114 @@ std::shared_ptr<afw::image::PhotoCalib> ConstrainedPhotometryModel::toPhotoCalib
+                                          << _chipVisitMap.bucket_count() << " buckets and a load factor of "
+                                          << _chipVisitMap.load_factor());
 }
 
-PhotometryMappingBase *ConstrainedPhotometryModel::findMapping(CcdImage const &ccdImage) const {
-    auto idMapping = _chipVisitMap.find(ccdImage.getHashKey());
-    if (idMapping == _chipVisitMap.end())
-        throw LSST_EXCEPT(pex::exceptions::InvalidParameterError,
-                          "ConstrainedPhotometryModel cannot find CcdImage " + ccdImage.getName());
-    return idMapping->second.get();
+double ConstrainedMagnitudeModel::transform(CcdImage const &ccdImage,
+                                            MeasuredStar const &measuredStar) const {
+    auto mapping = findMapping(ccdImage);
+    return mapping->transform(measuredStar, measuredStar.getInstMag());
+}
+
+double ConstrainedMagnitudeModel::transformError(CcdImage const &ccdImage,
+                                                 MeasuredStar const &measuredStar) const {
+    auto mapping = findMapping(ccdImage);
+    return mapping->transformError(measuredStar, measuredStar.getInstFlux(), measuredStar.getInstFluxErr());
+}
+
+std::shared_ptr<afw::image::PhotoCalib> ConstrainedMagnitudeModel::toPhotoCalib(
+        CcdImage const &ccdImage) const {
+    auto oldPhotoCalib = ccdImage.getPhotoCalib();
+    auto detector = ccdImage.getDetector();
+    auto ccdBBox = detector->getBBox();
+    ChipVisitPhotometryMapping *mapping = dynamic_cast<ChipVisitPhotometryMapping *>(findMapping(ccdImage));
+    // There should be no way in which we can get to this point and not have a ChipVisitMapping,
+    // so blow up if we don't.
+    assert(mapping != nullptr);
+    auto pixToFocal = detector->getTransform(afw::cameraGeom::PIXELS, afw::cameraGeom::FOCAL_PLANE);
+    // We know it's a Chebyshev transfo because we created it as such, so blow up if it's not.
+    auto visitTransfo =
+            std::dynamic_pointer_cast<PhotometryTransfoChebyshev>(mapping->getVisitMapping()->getTransfo());
+    assert(visitTransfo != nullptr);
+    auto focalBBox = visitTransfo->getBBox();
+
+    // Unravel our chebyshev coefficients to build an astshim::ChebyMap.
+    auto coeff_f = toChebyMapCoeffs(
+            std::dynamic_pointer_cast<PhotometryTransfoChebyshev>(mapping->getVisitMapping()->getTransfo()));
+    // Bounds are the bbox
+    std::vector<double> lowerBound = {focalBBox.getMinX(), focalBBox.getMinY()};
+    std::vector<double> upperBound = {focalBBox.getMaxX(), focalBBox.getMaxY()};
+
+    afw::geom::TransformPoint2ToGeneric chebyTransform(ast::ChebyMap(coeff_f, 1, lowerBound, upperBound));
+
+    using namespace std::string_literals;  // for oeprator""s to convert string literal->std::string
+    afw::geom::Transform<afw::geom::GenericEndpoint, afw::geom::GenericEndpoint> logTransform(
+            ast::MathMap(1, 1, {"y=pow(10.0,x/-2.5)"s}, {"x=-2.5*log10(y)"s}));
```

Comment:
In this particular case, we're building a AST Transform, so we have to do it internally there. We (@r-owen in particular) will probably request that a flux<->mag transform like this be added to AST itself, as it would likely be faster than this version.